apps/aecore/src/aec_peer_connection_listener.erl

```diff
@@ -6,57 +7,103 @@
 %%%=============================================================================
 -module(aec_peer_connection_listener).
 
--export([start_link/0, stop/0]).
+-behaviour(gen_server).
 
-%% TODO: Make this a gen_server
-start_link() ->
-    Pid = spawn_link(fun pc_listener/0),
-    register(?MODULE, Pid),
-    {ok, Pid}.
+%% API
+-export([ start_link/1
+        ]).
 
-stop() ->
-    ?MODULE ! stop,
-    ok.
+%% gen_server callbacks
+-export([ init/1
+        , handle_call/3
+        , handle_cast/2
+        , handle_info/2
+        , terminate/2
+        , code_change/3
+        ]).
 
-pc_listener() ->
+-define(SERVER, ?MODULE).
+
+-record(state, { lsock
+               , opts
+               , acceptor
+               }).
+
+%%====================================================================
+%% API functions
+%%====================================================================
+
+start_link([{sync_port, _}, {listen_address, _}] = Args) ->
```

Comment:
I am a bit confused about this use of proplists. The order of the tuple is not supposed to be relevant but with these matches it is. If the order is enforced I wouldn't go with a proplist, but a normal list because I think this is confusing. If you want to match why not use a map ?

Reply:
It is just me being (over) careful. Since the way gen_server (and friends) are passing arguments, I want to make sure we don't accidentally mix the args up. It is not a proplist, and should not be treated as such. It is a way of ensuring that the order of args is preserved. The call points are static (in the supervisor).

Reply:
I don't like using maps or records in cases like this either, since I want to ensure full match.

But I agree that it looks a bit strange ;-)